src/data_provider/src/ports/portSolarisHelper.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Wazuh SysInfo
+ * Copyright (C) 2015, Wazuh Inc.
+ * March 17, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stropts.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include <memory>
+
+#include "portSolarisHelper.hpp"
+
+int PortSolarisHelper::tcpConnEntrySize;
+int PortSolarisHelper::tcp6ConnEntrySize;
+int PortSolarisHelper::udpEntrySize;
+int PortSolarisHelper::udp6EntrySize;
+
+void PortSolarisHelper::mibGetItems(int sd, std::vector<mib_item_t>& items)
```

Comment:
If "items" is not required to be contiguous in memory, there are better STL containers to use rather than std::vector. This is because every time you push_back an element into a std::vector, the whole vector is reallocated to a new contiguous heap memory area including the copy operation of all its items.
Maybe std::deque would be a better STL container type.

Reply:
precisely in this case it is trying to emulate a vector C

Reply:
Where is it used or needed as C style vector?
I don't see it.

Reply:
My intention was not to stray too far from the original C code, but the change was made.